src/scanpy/preprocessing/_pca/_dask_sparse.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, cast, overload
+
+import numpy as np
+import scipy.linalg
+from numpy.typing import NDArray
+
+from .._utils import _get_mean_var
+
+if TYPE_CHECKING:
+    from typing import Literal
+
+    from numpy.typing import DTypeLike
+    from scipy import sparse
+
+    from ..._compat import DaskArray
+
+    CSMatrix = sparse.csr_matrix | sparse.csc_matrix
+
+
+@dataclass
+class PCASparseDask:
+    n_components: int | None = None
+
+    def fit(self, x: DaskArray) -> PCASparseFit:
+        # this method makes `self` into the fitted version
+        self.__class__ = PCASparseFit
```

Comment:
But why are these two different classes?

Reply:
Maybe add a comment?

Reply:
This is the only way I could found to have sklearn’s fit pattern work together with typing.

The idea is that before `fit`, all `trailing_dunder_` attributes don’t exist. Making them all `| None` and initializing them with `None` would be extremely ugly and isn’t accurate: after `fit`, they’re never `None`.

This makes that work: `PCASparseDask().fit()` returns a `PCASparseFit` instance which has all the `trailing_dunder_` attributes.

The only thing that doesn’t work is

```py
pca = PCASparseDask()
pca.fit()  # `pca` is actually a `PCASparseFit` now, but the type system doesn’t know
pca = pca.fit()  # this works
```

Reply:
I’ll happily add more comments, but I don’t know what you want explained in them, please elaborate :D

Reply:
1. I guess some condensed version of your entire previous comment would be a bad comment
2. Do you think it's possible this might affect code readability/debugging?  I'm having trouble reasoning what you mean by "type system doesn't know" - why else would we do this?

For example:

```python
class Foo:
   bar = "zar"

class Baz:
     def __init__(self):
         self.__class__ = Foo

assert not isinstance(Baz(), Baz)
```

This sort of thing seems a bit confusing to people who might be debugging

Reply:
True, but I think the point of this was just for readability i.e.,:

```python
type(Baz())
# Foo
```

If you feel like it's ok, I don't mind.